File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # ans = s[::-1]
        # res = ""

        # i = 0

        # while i < len(ans):
        #     while i < len(ans) and ans[i] == " ":
        #         i += 1
            
        #     word = ""
        #     j = i

        #     while j < len(ans) and ans[j] != " ":
        #         j += 1

        #     word += ans[i:j]
        #     word = word[::-1]
        #     res += " " + word
        #     i = j+1

        # return res.strip()

        return " ".join(s.split()[::-1])

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        reverse(s.begin(), s.end());

        string res = "";

        for(int i = 0; i < s.size(); i++){
            string word = "";
            while(i < s.size() && s[i] != ' '){
                word.push_back(s[i]);
                i++;
            }

            if(!word.empty()){
                reverse(word.begin(), word.end());
                res += ' ' + word;
            }
        }

        return res.substr(1);
    }
};